### ISSUE TITLE
Add support to stackLifecycle

### ISSUE THREAD

**Issue by szuecs** [zalando-incubator/stackset-controller#8]
Implement the field `stackLifecycle`

```
apiVersion: zalando.org/v1
kind: StackSet
metadata:
  name: my-app
spec:
  # optional Ingress definition.
  ingress:
    hosts: [my-app.example.org, alt.name.org]
  stackLifecycle:
    scaledownTTLSeconds: 300
    limit: 5
  stackTemplate:
   ....
```

stackLifecycle definition tells when stackset-controller should clean up the resources from and old stack that gets no traffic anymore.

- scaledownTTLSeconds: number of seconds to wait before scaling down the replicas of a stack to 0
- limit: at most N stacks should have no traffic

**Comment by linki** [comment]
@szuecs How should this work?

**Comment by szuecs** [comment]
I ported the issues, but I guess @mikkeloscar knows more :)

**Comment by mikkeloscar** [comment]
Fixed in #33
